src/schedule/ll_schedule.c

```diff
@@ -63,18 +63,29 @@ struct ll_queue_shared_context {
 
 static struct ll_queue_shared_context *ll_shared_ctx;
 
+static inline uint64_t ll_get_timer(struct ll_schedule_data *queue)
+{
+	return queue->ts->timer_get(&queue->ts->timer);
+}
+
 /* calculate next timeout */
 static inline uint64_t queue_calc_next_timeout(struct ll_schedule_data *queue,
 					       uint64_t start)
 {
-	return queue->ticks_per_msec * queue->timeout / 1000 + start;
-}
+	uint64_t ticks = queue->ticks_per_msec * queue->timeout / 1000 + start;
 
-static inline uint64_t ll_get_timer(struct ll_schedule_data *queue)
-{
-	return queue->ts->timer_get(&queue->ts->timer);
+	/* Did queued tasks took more time than scheduled period?
+	 * NOTE: We also take into account ticks needed to finish
+	 * rescheduling procedure ().
+	 */
+	if (ticks < ll_get_timer(queue) + TICKS_NEEDED_TO_FINISH_RESCHEDULE)
+		ticks = ll_get_timer(queue) +
```

Comment:
More efficient to do this
```c
temp_ticks = ll_get_timer(queue) + TICKS_NEEDED_;
if (ticks < temp_ticks)
    ticks = temp_ticks;
```

Reply:
Yes sure, there is no need to call ll_get_timer() twice.